examples/finance/monte_carlo.py

```diff
@@ -43,7 +43,9 @@
     call = discount_factor * call_sum / n_samples;
     ''',
     preamble='''
+    #ifndef __HIPCC__
```

Comment:
FYI in other places we use `__HIP_DEVICE_COMPILE__`, but the effect should be the same, so just a matter of style choice here.